File: C++/6. ZigZag Conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if(numRows==1)
            return s;
        string ans = "";
        int len = s.length();
        for(int i=0;i<numRows;i++)
        {
            int t1 = 2*i, t2 = 2*(numRows-1-i);
            int k = i;
            if(t1 == 0 || t2 == 0)
            {
                t1 = max(t1,t2);
                while(k<len)
                {
                    ans += s[k];
                    k += t1;
                }
            }
            else
            {
                while(k<len)
                {
                    ans += s[k];
                    k += t2;
                    if(k >= len)
                        break;
                    ans += s[k];
                    k += t1;
                }
            }
        }
        return ans;
    }
};

File: Python/6. ZigZag Conversion.py
class Solution:
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows == 1:
            return s
        ans = ""
        for i in range(numRows):
            t1, t2 = 2 * i, 2 * (numRows - i - 1)
            k = i
            if t1 == 0 or t2 == 0:
                t1 = max(t1, t2)
                while  k < len(s):
                    ans += s[k]
                    k += t1
            else:
                while k < len(s):
                    ans += s[k]
                    k += t2
                    if k >= len(s):
                        break
                    ans += s[k]
                    k += t1
        return ans